### PR TITLE
Topic recognition - Ziyan Wang (45758151) OASIS DCGAN

### PR DESCRIPTION
**Pull request by Ruaman**
# DCGAN On OASIS 
This project is an assignment of [COMP3710](https://my.uq.edu.au/programs-courses/course.html?course_code=comp3710) from The University of Queensland. The chosen topic is No.6 (OASIS)

## Author 
Ziyan Wang 45758151

## Script
The project includes the following three script files.
1. ImgLoader: This class is responsible for the load the dataset from image files and transfer to tensor object.
2. DCGAN: This class is responsible for the model creation, model training based on ImgLoader, and SSIM/evaluation.
3. main: The main file calls the function from the above two classes to run the experiments and present the results.

## Dependencies
1. Tensorflow
2. time
3. matplotlib
4. IPython
5. pathlib
6. PIL
7. numpy (data processing in ImgLoader only)

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages 
- Code checks out for hard difficulty 
- Results are good

Other Feedback
- Please gitignore IDE config files `.idea`
- Nice grouping of files

**Comment by rexsw** [comment]
Code looks good for hard difficulty

Remember to add a description to your pull request to explain what’s in the pull request.

Comments like 
   def discriminator_loss(self, real_output, fake_output):
        """
        A function that takes the output from generator and real image and returns the loss
Don’t add to the readability of the code and should be left out. However many other comments throughout the code are well done

Tensorflow has image functions built into it which you can use rather than requiring your project to have another library imported.

File: recognition/OASIS_DCGAN_Ziyan/src/DCGAN.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
+import os
```
Comment (shakes76) [inline recognition/OASIS_DCGAN_Ziyan/src/DCGAN.py:1]: Code header block should always be present on scripts

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. conflicting .gitignore file file blocking the merge.**
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Comment by Ruaman** [comment]
Conflicts fixed. The Request Description makes more sense
